bank_atm.rb

```diff
@@ -0,0 +1,116 @@
+$LOAD_PATH << '.'
+require 'validations'
+require_relative 'models/initializer'
+
+#Begin of demo data
+p1 = Person.new("John", "Developer", "j@yahoo.com")
+p2 = Person.new("Tudor", "Climbing instructor", "t@tahoo.com")
+
+persons = []
+persons << p1
+persons << p2
+
+b1 = BankAccount.new("1234", p1, 150)
+b2 = BankAccount.new("5555", p1, 5500)# with 1 widthrawal already
+b3 = BankAccount.new("9999", p2, 7000)
+
+b2.withdraw(4000)
+#End of demo data
+
+#Auxiliary methods:
+def validate_person_name(name, persons)
+    person = persons.find { |person| person.name == name }
+    raise StandardError.new("No user found with that name") if person.nil?
+    person
+end
+
+def validate_person_bank_account(bank_accounts, pin)
+    selected_bank_acc = bank_accounts.find { |account| account.pin_matches?(pin) }
+    raise StandardError.new "Invalid PIN" if selected_bank_acc.nil?
+    selected_bank_acc
+end
+
+#Options methods:
+def deposit_funds(persons)
```

Comment:
you could avoid always passing the `persons` as an argument by setting the persons as an instance variable